scripts/manual_validation_data_processing.R

```
@@ -0,0 +1,135 @@
+               col1_position = "recruitment_status_reg1", col2_position = "recruitment_status_reg2") |>
+  flip_columns("has_summary_results_reg1_main", "has_summary_results_reg2_main", 
+               col1_position = "overall_recruitment_status_reg2", col2_position = "has_summary_results_reg1_main_new") |>
+  flip_columns("has_summary_results_reg1_sensitivity", "has_summary_results_reg2_sensitivity", 
+               col1_position = "has_summary_results_reg2_main", col2_position = "has_summary_results_reg1_sensitivity_new")
+
+#### QUALITY CHECK OF TRANSFORMATION ####
+# The idea is to compare the columns from data_transformed with the columns from drks_ctgov_data (before transformation)
+
+#CREATE FUNCTION FOR QUALITY CHECK
+compare_columns <- function(data1, col1, data2, col2) {
+  if (setequal(data1[[col1]], data2[[col2]])) {
+    message(sprintf("Columns %s and %s contain the same values", col1, col2))
+  } else {
+    message(sprintf("Columns %s and %s do NOT contain the same values", col1, col2))
+  }
+}
+
+#APPLYING FUNCTION
+#define column pairs
+column_pairs <- list(
+  c("trn1", "trn2"),
+  c("registry1", "registry2"),
+  c("completion_date_reg1", "completion_date_reg2"),
+  c("completion_month_year_reg1", "completion_month_year_reg2"),
+  c("completion_date_type_reg1", "completion_date_type_reg2"),
+  c("recruitment_status_reg1", "recruitment_status_reg2"),
+  c("overall_recruitment_status_reg1", "overall_recruitment_status_reg2"),
+  c("has_summary_results_reg1_main", "has_summary_results_reg2_main"),
+  c("has_summary_results_reg1_sensitivity", "has_summary_results_reg2_sensitivity")
+)
+
+#perform comparisons
+for (pair in column_pairs) {
+  compare_columns(data_transformed, pair[1], drks_ctgov_data, pair[2])
+  compare_columns(data_transformed, pair[2], drks_ctgov_data, pair[1])
+}
+
+#### RE-JOINING FINAL DATASET FOR ANALYSIS ####
+joined_data <- bind_rows(euctr_data, data_transformed)
```

@elisabascunan I'm getting white spaces in some columns, which actually created an issue in some of Luciano's code so we should fix it. In general it's best to avoid any white spaces as it can lead to silent errors. Do you know how it got introduced?

See below a suggestion for how to handle the white spaces. I see white spaces for `trn1`, `trn2`, `recruitment_status_reg1` and `recruitment_status_reg2`, but please check if elsewhere. Btw, this is also where you could select and order the columns (instead of relocating earlier, see comment above).

```
joined_data <- bind_rows(euctr_data, data_transformed) |>
  mutate(
    # remove leading and trailing white spaces
    trn1 = trn1 |> trimws(),
    trn2 = trn2 |> trimws(),
    recruitment_status_reg1 = recruitment_status_reg1 |> trimws(),
    recruitment_status_reg2 = recruitment_status_reg2 |> trimws()
    ) |>
  select(trn1,
         trn2,
         registry1,
         ...
         )
```

---

It seems that with the new file (`manual_validation_raw.xlsx`) this issue is solved for trn1 and trn2. For `recruitment_status_reg1`, `recruitment_status_reg2,` and any comment columns there are gonna be white spaces because there are more than two words as possible values (example: "Prematurely Ended").